### PR TITLE
Add `QuantizedVariableSpec` to `QuantizedVariable`

### PR DESCRIPTION
**Pull request by lgeiger**
This adds a trace type to our quantized variable according to https://github.com/keras-team/keras/commit/ed99e34f279a2d2d6a44af87ee64f8fc98c7e8b9

This fixes an issue I am seeing on single GPU with optimizers that use `jit_compile=True`.

### CODE REVIEW

File: larq/quantized_variable.py [hunk 1]
```diff
@@ -330,6 +366,11 @@ def shape(self):
     def get_shape(self):
         return self.latent_variable.get_shape()
 
+    def __tf_tracing_type__(self, context):
+        if _SUPPORTS_TRACE_TYPE:
+            return QuantizedVariableSpec(self)
+        return NotImplemented
```
Comment (DzedCPT) [inline larq/quantized_variable.py:372]: It's safe to return `NotImplemented` here? What does the caller do with the result?

**Comment by DzedCPT** [comment]
> This adds a trace type to our quantized variable according to [keras-team/keras@ed99e34](https://github.com/keras-team/keras/commit/ed99e34f279a2d2d6a44af87ee64f8fc98c7e8b9)

Can we steal the test from that PR as well?

**Review by DzedCPT — APPROVED** [review]
Not super familiar with this code so take my approval with a pinch of salt, but it looks reasonable to me.

**Comment by lgeiger** [comment]
> Can we steal the test from that PR as well?

Checking the dtypes doesn't really apply to us here. But I was able to make a minimal reproduction that would crash the tf function tracing without this fix.  I added it as a test in 983decb885d7f81247167f65de6da86461a84a0c

File: larq/quantized_variable.py (hunk 1 above)
Comment (lgeiger) [reply larq/quantized_variable.py:372]: I think so, yes. If the TensorFlow version doesn't support trace type I am pretty sure this function will never get called so it's fine to return `NotImplemented`.

**Pull request merged**